### ISSUE TITLE
Toggle keyboard nav feels inconsistent

### ISSUE THREAD

**Issue by VanWeapon** [arnoudkooi/ServiceNow-Utils#478]
Update introduced in this commit: 3264094576d90a80d9fcf59afbb01f002bfe4b28
If the intent is to have SNU remember the state of the toggle, this doesn't appear to be happening reliably. 
Personally I would prefer this toggle being a setting because I would never turn it off. Having it off by default is also very jarring since for me pressing ctrl+1 (which is basically muscle memory at this point) jumps me to the first tab and I lose where I was

**Comment by VanWeapon** [comment]
Sure, In this example I'm using `/u` to search in the user table 
By default the toggle is off, so if i hit `1` it will add  '1' to the slashcommand search input instead of opening in the current window, if i hit `Ctrl+1` which would force the open in a new tab, instead my browser shortcuts activate, and `Ctrl+[num]` is a shortcut on most browsers to go to the tab at that num index, i.e. the first tab.

![image](https://github.com/arnoudkooi/ServiceNow-Utils/assets/30616047/c056e3ef-e640-4cc4-b6f3-a116bb6a85be)

When using a slashcommand like `/u` which returns a list, the numbered keys to navigate a selection don't show up until you 'execute' the search by hitting right arrow anyway, so I don't see how entering numbers as part of the search was a problem before:

e.g., entering a search term before hitting right arrow, window navigation not active
![image](https://github.com/arnoudkooi/ServiceNow-Utils/assets/30616047/426eeaf6-6ffa-48b1-9598-bc13a6bfa3c1)


I get why the feature is there for menus/favourites/history, but for me it seems like the toggle keeps turning itself back to off by default leading to cases where I hit `Ctrl+1` or whatever and it changes my tab instead of opening the target.

**Comment by arnoudkooi** [comment]
Is this Firefox? State is stored in `localStorage.getItem('snunumbernav');`

**Comment by VanWeapon** [comment]
Opera, happened again this morning

**Comment by arnoudkooi** [comment]
Have you used other browsers where it happens? I noticed in Opera the new sidepanel for InstanceTag doesn't work, need to look into it...

**Comment by VanWeapon** [comment]
its strange, the way the code is written it looks like the default should be true actually, but for me it just keeps turning itself off. 
If i write my own inject which manually sets the localStorage item to true, then the feature breaks and toggling with shift doesn't work anymore. 

This looks like if there's no value it should initialise as true? 
https://github.com/arnoudkooi/ServiceNow-Utils/commit/3264094576d90a80d9fcf59afbb01f002bfe4b28#diff-8dcb72b9c92c5fa9eb8602ce15b90717f8717699b5865db86078df006be5c4c0R5270-R5279

When i load up Edge with the extension, the initial state is off, but it appears as though the toggle keeps it state. Its hard to replicate though because sometimes the state will keep itself on all day, other times like yesterday i had to toggle it back on manually like 4 or 5 times. 

Maybe the Shift event listener here is being picked up even if the slashcommand window isn't open? 
https://github.com/arnoudkooi/ServiceNow-Utils/commit/3264094576d90a80d9fcf59afbb01f002bfe4b28#diff-8dcb72b9c92c5fa9eb8602ce15b90717f8717699b5865db86078df006be5c4c0R606-R614

I'm going to try seeing if I can keep track of the localStorage changes throughout the day and see if that's what's happening.

**Comment by arnoudkooi** [comment]
Thanks for diving into this.
There is a "[storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event)" event that could help monitor any changes.

**Comment by VanWeapon** [comment]
Aha, i found it 

The event listener is always active when the slashcommand popup is open, not just when the prompt is there to toggle. 

When typing `_` e.g. to type a table name `/change_task` the hitting underscore (shift + -) is causing the toggle to fire

**Comment by arnoudkooi** [comment]
Makes sense, I'll add a check so it only toggles when links are visible

```javascript
if (e.key == 'Shift' && window.top.document.querySelectorAll('div.snutils span.dispidx').length)

**Comment by arnoudkooi** [comment]
This is published in 7.4.0.9, hope that reliefs the issue

**Comment by VanWeapon** [comment]
I think this is good now, haven't had any issues recently
